app/scripts/modules/kubernetes/src/v2/pipelines/stages/patchManifest/PatchManifestStageForm.tsx

```diff
@@ -0,0 +1,174 @@
+  IPatchManifestStageConfigFormProps & IFormikStageConfigInjectedProps,
+  IPatchManifestStageConfigFormState
+> {
+  public readonly textSource = 'text';
+  public readonly artifactSource = 'artifact';
+  private readonly excludedManifestArtifactTypes = [
+    ArtifactTypePatterns.DOCKER_IMAGE,
+    ArtifactTypePatterns.KUBERNETES,
+    ArtifactTypePatterns.FRONT50_PIPELINE_TEMPLATE,
+    ArtifactTypePatterns.EMBEDDED_BASE64,
+    ArtifactTypePatterns.MAVEN_FILE,
+  ];
+
+  public constructor(props: IPatchManifestStageConfigFormProps & IFormikStageConfigInjectedProps) {
+    super(props);
+    const patchBody: string = get(props.formik.values, 'patchBody');
+    const isTextManifest: boolean = get(props.formik.values, 'source') === this.textSource;
+    this.state = {
+      rawManifest: !isEmpty(patchBody) && isTextManifest ? yamlDocumentsToString([patchBody]) : '',
+    };
+  }
+
+  private onManifestArtifactSelected = (expectedArtifactId: string): void => {
+    this.props.formik.setFieldValue('manifestArtifactId', expectedArtifactId);
+    this.props.formik.setFieldValue('manifestArtifact', null);
+  };
+
+  private onManifestArtifactEdited = (artifact: IArtifact) => {
+    this.props.formik.setFieldValue('manifestArtifactId', null);
+    this.props.formik.setFieldValue('manifestArtifact', artifact);
+  };
+
+  private onManifestArtifactAccountSelected = (accountName: string): void => {
+    this.props.formik.setFieldValue('manifestArtifactAccount', accountName);
+  };
+
+  private getRequiredArtifacts = (): IManifestBindArtifact[] => {
+    const { requiredArtifactIds, requiredArtifacts } = this.props.formik.values;
+    return (requiredArtifactIds || [])
+      .map((id: string) => ({ expectedArtifactId: id }))
+      .concat(requiredArtifacts || []);
+  };
+
+  private onRequiredArtifactsChanged = (bindings: IManifestBindArtifact[]): void => {
+    this.props.formik.setFieldValue(
+      'requiredArtifactIds',
+      bindings.filter(b => b.expectedArtifactId).map(b => b.expectedArtifactId),
+    );
+    this.props.formik.setFieldValue('requiredArtifacts', bindings.filter(b => b.artifact));
+  };
+
+  private handleRawManifestChange = (rawManifest: string, manifests: any): void => {
+    this.setState({
+      rawManifest,
+    });
+    this.props.formik.setFieldValue('patchBody', manifests[0]);
+  };
+
+  private onManifestSelectorChange = (): void => {};
+
+  private getSourceOptions = (): Array<Option<string>> => {
+    return map([this.textSource, this.artifactSource], option => ({
+      label: capitalize(option),
+      value: option,
+    }));
+  };
+
+  public render() {
+    const stage = this.props.formik.values;
+    return (
+      <div className="container-fluid form-horizontal">
+        <h4>Resource to Patch</h4>
+        <ManifestSelector
+          application={this.props.application}
+          modes={[SelectorMode.Static, SelectorMode.Dynamic]}
+          onChange={this.onManifestSelectorChange}
```

Comment:
looks like this is a noop, how does that work?

Reply:
i can't understand why the event handler is even necessary. i've tested it to the best of my knowledge and it appears that model is correctly updated without anything further.